src/circuit_proof/r1cs.rs

```diff
@@ -66,15 +68,64 @@ pub struct ConstraintSystem {
     v_assignments: Vec<Assignment>,
 }
 
-impl ConstraintSystem {
-    pub fn new() -> Self {
-        ConstraintSystem {
+impl<'a> ConstraintSystem<'a> {
+    pub fn prover_new(
+        transcript: &'a mut Transcript,
+        // TODO: encapsulate v, v_blinding, pedersen_gens into one "commitment" struct
+        v: Vec<Scalar>,
+        v_blinding: Vec<Scalar>,
+        pedersen_gens: PedersenGenerators,
+    ) -> (Self, Vec<Variable>, Vec<RistrettoPoint>) {
+        let mut cs = ConstraintSystem {
+            transcript,
+            constraints: vec![],
+            aL_assignments: vec![],
+            aR_assignments: vec![],
+            aO_assignments: vec![],
+            v_assignments: vec![],
```

Comment:
Where does `v_assignments` get initialised?

Reply:
Oh, does that happen via the calls to `assign_committed`?

Reply:
Here, the `v_assignments` gets initialized in line 102, in the `cs.assign_committed()` call